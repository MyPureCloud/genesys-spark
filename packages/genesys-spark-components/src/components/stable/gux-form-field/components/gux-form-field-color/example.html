<h1>gux-form-field-color</h1>

<p>slot a input[type=color] element into a gux-form-field-color element</p>

<h2>Examples</h2>
<form onchange="notify(event)" oninput="notify(event)">
  <fieldset>
    <legend>Label Position</legend>

    <gux-form-field-color>
      <input slot="input" type="color" name="lp-1" value="#ff4f1f" />
      <label slot="label">Default</label>
    </gux-form-field-color>

    <gux-form-field-color>
      <input slot="input" type="color" name="lp-2" value="#ff4f1f" />
      <label slot="label">Default long label</label>
    </gux-form-field-color>

    <gux-form-field-color label-position="above">
      <input slot="input" type="color" name="lp-3" value="#ff4f1f" />
      <label slot="label">Above</label>
    </gux-form-field-color>

    <gux-form-field-color label-position="beside">
      <input slot="input" type="color" name="lp-4" value="#ff4f1f" />
      <label slot="label">Beside</label>
    </gux-form-field-color>

    <gux-form-field-color label-position="screenreader">
      <input slot="input" type="color" name="lp-5" value="#ff4f1f" />
      <label slot="label">Screenreader</label>
    </gux-form-field-color>
  </fieldset>

  <fieldset>
    <legend>Input attributes</legend>

    <gux-form-field-color>
      <input slot="input" type="color" name="d-1" value="#203b73" disabled />
      <label slot="label">True</label>
    </gux-form-field-color>

    <gux-form-field-color>
      <input slot="input" type="color" name="r-1" value="#8452cf" required />
      <label slot="label">True</label>
    </gux-form-field-color>
  </fieldset>

  <fieldset>
    <legend>Error</legend>

    <gux-form-field-color>
      <input slot="input" type="color" name="e-1" value="#cc3ebe" />
      <label slot="label">Default</label>
      <span slot="error">This is an error message</span>
    </gux-form-field-color>
  </fieldset>

  <fieldset>
    <legend>Help</legend>

    <gux-form-field-color>
      <input slot="input" type="color" name="e-1" value="#cc3ebe" />
      <label slot="label">Default</label>
      <span slot="help">This is a help message</span>
    </gux-form-field-color>
  </fieldset>

  <fieldset>
    <legend>Label Info</legend>

    <gux-form-field-color>
      <input slot="input" type="color" name="e-1" value="#cc3ebe" />
      <label slot="label">Default</label>
      <gux-label-info-beta slot="label-info">
        <span slot="content">This is some tooltip text</span>
      </gux-label-info-beta>
    </gux-form-field-color>
  </fieldset>

  <fieldset>
    <legend>Optional Indicator Mark</legend>

    <gux-form-field-color indicator-mark="optional">
      <input slot="input" type="color" name="e-1" value="#cc3ebe" />
      <label slot="label">Default</label>
    </gux-form-field-color>

    <gux-form-field-color indicator-mark="optional">
      <input slot="input" type="color" name="e-1" value="#cc3ebe" required />
      <label slot="label">Default</label>
    </gux-form-field-color>
  </fieldset>
</form>

<style>
  gux-form-field-color {
    margin: 10px;
  }
</style>
