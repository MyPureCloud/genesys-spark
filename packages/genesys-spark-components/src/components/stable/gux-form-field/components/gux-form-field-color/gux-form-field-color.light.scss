gux-form-field-color {
  input {
    &:focus,
    &:focus-visible {
      border: 0;
      outline: none;
      box-shadow: none;
    }

    &::-webkit-color-swatch-wrapper {
      padding: 0;
    }

    &::-webkit-color-swatch,
    &::-moz-color-swatch {
      width: var(--gse-ui-colorPicker-input-swatchSizing);
      height: var(--gse-ui-colorPicker-input-swatchSizing);
      border: none;
      border-radius: var(--gse-ui-colorPicker-input-swatchBorderRadius);
    }
  }
}
