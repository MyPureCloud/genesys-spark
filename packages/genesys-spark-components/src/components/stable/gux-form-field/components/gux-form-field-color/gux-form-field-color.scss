@use '../../gux-form-field.scss';
@use '../../functional-components/gux-form-field-container/gux-form-field-container.scss';
@use '../../functional-components/gux-form-field-error/gux-form-field-error.scss';
@use '../../functional-components/gux-form-field-label/gux-form-field-label.scss';
@use '../../functional-components/gux-form-field-help/gux-form-field-help.scss';

@include gux-form-field-container.Style;
@include gux-form-field-error.Style;
@include gux-form-field-label.Style;
@include gux-form-field-help.Style;

:host {
  display: block;
}

::slotted(label) {
  @include gux-form-field.gse-ui-formControl-label-bold;
}

::slotted(input) {
  flex: 0 1 auto;
  align-self: auto;
  order: 0;
  width: var(--gse-ui-colorPicker-input-swatchSizing);
  min-width: var(--gse-ui-colorPicker-input-swatchSizing);
  height: var(--gse-ui-colorPicker-input-swatchSizing);
  padding: 0;
  overflow: hidden;
  font-size: var(--gse-ui-formControl-input-contentText-fontSize);
  color: var(--gse-ui-formControl-input-populatedColor);
  text-align: right;
  text-overflow: ellipsis;
  background-color: var(--gse-ui-formControl-input-backgroundColor);
  border: none;
  outline: none;

  &::placeholder {
    color: var(--gse-ui-formControl-input-placeholderColor);
    opacity: 1; // TODO:  COMUI-2187
  }
}

.gux-input-and-error-container {
  flex-grow: 1;

  .gux-input {
    display: flex;
    flex-direction: row;
    flex-wrap: nowrap;
    align-content: stretch;
    align-items: center;

    .gux-input-container {
      box-sizing: border-box;
      display: flex;
      flex-direction: row;
      flex-wrap: nowrap;
      gap: var(--gse-ui-formControl-input-gap);
      place-content: stretch center;
      align-items: center;
      align-self: auto;
      order: 0;
      width: var(--gse-ui-formControl-input-colorPicker-size);
      height: var(--gse-ui-formControl-input-colorPicker-size);
      font-family: var(--gse-ui-formControl-input-contentText-fontFamily);
      font-size: var(--gse-ui-formControl-input-contentText-fontSize);
      line-height: var(--gse-ui-formControl-input-contentText-lineHeight);
      color: var(--gse-ui-formControl-input-populatedColor);
      background-color: var(--gse-ui-formControl-input-backgroundColor);
      background-image: none;
      border: var(--gse-ui-formControl-input-default-border-width)
        var(--gse-ui-formControl-input-default-border-style)
        var(--gse-ui-formControl-input-default-border-color);
      border-radius: var(--gse-ui-formControl-input-borderRadius);

      &.gux-disabled {
        opacity: var(--gse-ui-formControl-input-disabled-opacity);
      }

      &:focus-within {
        border: var(--gse-ui-formControl-input-active-border-width)
          var(--gse-ui-formControl-input-active-border-style)
          var(--gse-ui-formControl-input-active-border-color);
        border-radius: var(--gse-ui-formControl-focusRing-borderRadius);
        outline: var(--gse-ui-formControl-input-focus-border-width)
          var(--gse-ui-formControl-input-focus-border-style)
          var(--gse-ui-formControl-input-focus-border-color);
        outline-offset: var(--gse-semantic-focusRing-offset);
      }

      &:hover {
        cursor: pointer;
        border-color: var(--gse-ui-formControl-input-hover-border-color);
      }
    }

    &.gux-input-error {
      .gux-input-container {
        border-color: var(--gse-ui-formControl-input-error-border-color);
      }
    }
  }
}
