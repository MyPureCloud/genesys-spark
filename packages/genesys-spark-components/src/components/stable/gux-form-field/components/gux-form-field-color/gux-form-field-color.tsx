import { Component, Element, h, JSX, Prop, State } from '@stencil/core';

import { calculateInputDisabledState } from '@utils/dom/calculate-input-disabled-state';
import { onInputDisabledStateChange } from '@utils/dom/on-input-disabled-state-change';
import { OnMutation } from '@utils/decorator/on-mutation';
import { onRequiredChange } from '@utils/dom/on-attribute-change';
import { preventBrowserValidationStyling } from '@utils/dom/prevent-browser-validation-styling';
import { hasSlot } from '@utils/dom/has-slot';

import {
  GuxFormFieldHelp,
  GuxFormFieldError,
  GuxFormFieldLabel,
  GuxFormFieldContainer
} from '../../functional-components/functional-components';

import {
  GuxFormFieldIndicatorMark,
  GuxFormFieldLabelPosition
} from '../../gux-form-field.types';
import {
  getComputedLabelPosition,
  validateFormIds,
  getSlottedInput
} from '../../gux-form-field.service';
import { trackComponent } from '@utils/tracking/usage';

/**
 * @slot input - Required slot for input tag
 * @slot label - Required slot for label tag
 * @slot error - Optional slot for error message
 * @slot help - Optional slot for help message
 */
@Component({
  styleUrl: 'gux-form-field-color.scss',
  tag: 'gux-form-field-color',
  shadow: true
})
export class GuxFormFieldColor {
  private input: HTMLInputElement;
  private label: HTMLLabelElement;
  private disabledObserver: MutationObserver;
  private requiredObserver: MutationObserver;

  @Element()
  private root: HTMLElement;

  @Prop()
  labelPosition: GuxFormFieldLabelPosition;

  /**
   * Field indicator mark which can show *, (optional) or blank
   * Defaults to required. When set to required, the component will display * for required fields and blank for optional
   * When set to optional, the component will display (optional) for optional and blank for required.
   */
  @Prop()
  indicatorMark: GuxFormFieldIndicatorMark = 'required';

  @State()
  private computedLabelPosition: GuxFormFieldLabelPosition;

  @State()
  private disabled: boolean;

  @State()
  private required: boolean;

  @State()
  private hasError: boolean = false;

  @State()
  private hasHelp: boolean = false;

  @OnMutation({ childList: true, subtree: true })
  onMutation(): void {
    this.hasError = hasSlot(this.root, 'error');
    this.hasHelp = hasSlot(this.root, 'help');
  }

  componentWillLoad(): void {
    this.setInput();
    this.setLabel();

    this.hasError = hasSlot(this.root, 'error');
    this.hasHelp = hasSlot(this.root, 'help');

    trackComponent(this.root, { variant: this.variant });
  }

  disconnectedCallback(): void {
    if (this.disabledObserver) {
      this.disabledObserver.disconnect();
    }
    if (this.requiredObserver) {
      this.requiredObserver.disconnect();
    }
  }

  render(): JSX.Element {
    return (
      <GuxFormFieldContainer labelPosition={this.computedLabelPosition}>
        <GuxFormFieldLabel
          position={this.computedLabelPosition}
          required={this.required}
          indicatorMark={this.indicatorMark}
        >
          <slot name="label" onSlotchange={() => this.setLabel()} />
        </GuxFormFieldLabel>
        <div class="gux-input-and-error-container">
          <div
            class={{
              'gux-input': true,
              'gux-input-error': this.hasError
            }}
          >
            <div
              class={{
                'gux-input-container': true,
                'gux-disabled': this.disabled
              }}
            >
              <slot name="input" onSlotchange={() => this.setInput()} />
            </div>
          </div>
          <GuxFormFieldError show={this.hasError}>
            <slot name="error" />
          </GuxFormFieldError>
          <GuxFormFieldHelp show={!this.hasError && this.hasHelp}>
            <slot name="help" />
          </GuxFormFieldHelp>
        </div>
      </GuxFormFieldContainer>
    ) as JSX.Element;
  }

  private get variant(): string {
    return this.labelPosition ? this.labelPosition.toLowerCase() : 'none';
  }

  private setInput(): void {
    this.input = getSlottedInput(
      this.root,
      'input[type="color"][slot="input"]'
    );

    preventBrowserValidationStyling(this.input);

    this.disabled = calculateInputDisabledState(this.input);
    this.required = this.input.required;

    this.disabledObserver = onInputDisabledStateChange(
      this.input,
      (disabled: boolean) => {
        this.disabled = disabled;
      }
    );
    this.requiredObserver = onRequiredChange(
      this.input,
      (required: boolean) => {
        this.required = required;
      }
    );

    validateFormIds(this.root, this.input);
  }

  private setLabel(): void {
    this.label = this.root.querySelector('label[slot="label"]');

    this.computedLabelPosition = getComputedLabelPosition(
      this.label,
      this.labelPosition
    );
  }
}
