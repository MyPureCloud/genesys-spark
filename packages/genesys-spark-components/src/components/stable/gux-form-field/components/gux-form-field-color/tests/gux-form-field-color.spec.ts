import { newSpecPage } from '@test/specTestUtils';

import { GuxFormFieldColor } from '../gux-form-field-color';

const components = [GuxFormFieldColor];
const language = 'en';

describe('gux-form-field-color', () => {
  it('should build', async () => {
    const html = `
      <gux-form-field-color>
        <input slot="input" type="color" value="#ff0000"/>
        <label slot="label">Label</label>
        <span slot="error">Error message</span>
      </gux-form-field-color>
    `;
    const page = await newSpecPage({ components, html, language });

    expect(page.rootInstance).toBeInstanceOf(GuxFormFieldColor);
  });

  describe('#render', () => {
    describe('label-position', () => {
      [
        '',
        'label-position="above"',
        'label-position="beside"',
        'label-position="screenreader"'
      ].forEach((componentAttribute, index) => {
        it(`should render component as expected (${index + 1})`, async () => {
          const html = `
            <gux-form-field-color ${componentAttribute}>
              <input slot="input" type="color" value="#ff0000"/>
              <label slot="label">Label</label>
            </gux-form-field-color>
          `;
          const page = await newSpecPage({ components, html, language });

          expect(page.root).toMatchSnapshot();
        });
      });
    });

    describe('input attributes', () => {
      ['', 'disabled', 'required'].forEach((inputAttribute, index) => {
        it(`should render component as expected (${index + 1})`, async () => {
          const html = `
            <gux-form-field-color>
              <input slot="input" type="color" value="#ff0000" ${inputAttribute}/>
              <label slot="label">Label</label>
            </gux-form-field-color>
          `;
          const page = await newSpecPage({ components, html, language });

          expect(page.root).toMatchSnapshot();
        });
      });
    });

    describe('indicator marks', () => {
      ['', 'indicator-mark="optional"', 'indicator-mark="required"'].forEach(
        (indicatorMark, index) => {
          it(`should render component as expected (${index + 1})`, async () => {
            const html = `
              <gux-form-field-color ${indicatorMark}>
                <input slot="input" type="color" value="#ff0000" />
                <label slot="label">Label</label>
              </gux-form-field-color>
            `;
            const page = await newSpecPage({ components, html, language });

            expect(page.root).toMatchSnapshot();
          });
        }
      );
    });

    describe('help', () => {
      it('should render component as expected', async () => {
        const html = `
        <gux-form-field-color>
        <input slot="input" type="color" name="e-1" value="#cc3ebe" />
        <label slot="label">Default</label>
        <span slot="help">This is a help message</span>
      </gux-form-field-color>
        `;
        const page = await newSpecPage({ components, html, language });

        expect(page.root).toMatchSnapshot();
      });
    });

    it('should render the label info when provided', async () => {
      const html = `
            <gux-form-field-color>
              <input slot="input" type="color" name="e-1" value="#cc3ebe" />
              <label slot="label">Default</label>
              <gux-label-info-beta slot="label-info">
                <span slot="content">This is some tooltip text</span>
              </gux-label-info-beta>
            </gux-form-field-color>
          `;

      const page = await newSpecPage({ components, html, language });

      expect(page.root).toMatchSnapshot();
    });
  });
});
