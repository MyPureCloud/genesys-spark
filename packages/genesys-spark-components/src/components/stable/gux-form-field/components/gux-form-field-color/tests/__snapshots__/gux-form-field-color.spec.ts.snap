// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-form-field-color #render help should render component as expected 1`] = `
<gux-form-field-color>
  <mock:shadow-root>
    <div class="gux-above gux-form-field-container">
      <div class="gux-above gux-form-field-label">
        <slot name="label"></slot>
        <gux-form-field-label-indicator variant="required"></gux-form-field-label-indicator>
        <slot name="label-info"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-input-container">
            <slot name="input"></slot>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help gux-show">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <input aria-describedby="gux-form-field-help-i" id="gux-form-field-input-i" name="e-1" slot="input" type="color" value="#cc3ebe">
  <label for="gux-form-field-input-i" slot="label">
    Default
  </label>
  <span id="gux-form-field-help-i" slot="help">
    This is a help message
  </span>
</gux-form-field-color>
`;

exports[`gux-form-field-color #render indicator marks should render component as expected (1) 1`] = `
<gux-form-field-color>
  <mock:shadow-root>
    <div class="gux-above gux-form-field-container">
      <div class="gux-above gux-form-field-label">
        <slot name="label"></slot>
        <gux-form-field-label-indicator variant="required"></gux-form-field-label-indicator>
        <slot name="label-info"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-input-container">
            <slot name="input"></slot>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <input id="gux-form-field-input-i" slot="input" type="color" value="#ff0000">
  <label for="gux-form-field-input-i" slot="label">
    Label
  </label>
</gux-form-field-color>
`;

exports[`gux-form-field-color #render indicator marks should render component as expected (2) 1`] = `
<gux-form-field-color indicator-mark="optional">
  <mock:shadow-root>
    <div class="gux-above gux-form-field-container">
      <div class="gux-above gux-form-field-label">
        <slot name="label"></slot>
        <gux-form-field-label-indicator variant="optional"></gux-form-field-label-indicator>
        <slot name="label-info"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-input-container">
            <slot name="input"></slot>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <input id="gux-form-field-input-i" slot="input" type="color" value="#ff0000">
  <label for="gux-form-field-input-i" slot="label">
    Label
  </label>
</gux-form-field-color>
`;

exports[`gux-form-field-color #render indicator marks should render component as expected (3) 1`] = `
<gux-form-field-color indicator-mark="required">
  <mock:shadow-root>
    <div class="gux-above gux-form-field-container">
      <div class="gux-above gux-form-field-label">
        <slot name="label"></slot>
        <gux-form-field-label-indicator variant="required"></gux-form-field-label-indicator>
        <slot name="label-info"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-input-container">
            <slot name="input"></slot>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <input id="gux-form-field-input-i" slot="input" type="color" value="#ff0000">
  <label for="gux-form-field-input-i" slot="label">
    Label
  </label>
</gux-form-field-color>
`;

exports[`gux-form-field-color #render input attributes should render component as expected (1) 1`] = `
<gux-form-field-color>
  <mock:shadow-root>
    <div class="gux-above gux-form-field-container">
      <div class="gux-above gux-form-field-label">
        <slot name="label"></slot>
        <gux-form-field-label-indicator variant="required"></gux-form-field-label-indicator>
        <slot name="label-info"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-input-container">
            <slot name="input"></slot>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <input id="gux-form-field-input-i" slot="input" type="color" value="#ff0000">
  <label for="gux-form-field-input-i" slot="label">
    Label
  </label>
</gux-form-field-color>
`;

exports[`gux-form-field-color #render input attributes should render component as expected (2) 1`] = `
<gux-form-field-color>
  <mock:shadow-root>
    <div class="gux-above gux-form-field-container">
      <div class="gux-above gux-form-field-label">
        <slot name="label"></slot>
        <gux-form-field-label-indicator variant="required"></gux-form-field-label-indicator>
        <slot name="label-info"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-disabled gux-input-container">
            <slot name="input"></slot>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <input disabled="" id="gux-form-field-input-i" slot="input" type="color" value="#ff0000">
  <label for="gux-form-field-input-i" slot="label">
    Label
  </label>
</gux-form-field-color>
`;

exports[`gux-form-field-color #render input attributes should render component as expected (3) 1`] = `
<gux-form-field-color>
  <mock:shadow-root>
    <div class="gux-above gux-form-field-container">
      <div class="gux-above gux-form-field-label gux-required">
        <slot name="label"></slot>
        <gux-form-field-label-indicator required="" variant="required"></gux-form-field-label-indicator>
        <slot name="label-info"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-input-container">
            <slot name="input"></slot>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <input id="gux-form-field-input-i" required="" slot="input" type="color" value="#ff0000">
  <label for="gux-form-field-input-i" slot="label">
    Label
  </label>
</gux-form-field-color>
`;

exports[`gux-form-field-color #render label-position should render component as expected (1) 1`] = `
<gux-form-field-color>
  <mock:shadow-root>
    <div class="gux-above gux-form-field-container">
      <div class="gux-above gux-form-field-label">
        <slot name="label"></slot>
        <gux-form-field-label-indicator variant="required"></gux-form-field-label-indicator>
        <slot name="label-info"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-input-container">
            <slot name="input"></slot>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <input id="gux-form-field-input-i" slot="input" type="color" value="#ff0000">
  <label for="gux-form-field-input-i" slot="label">
    Label
  </label>
</gux-form-field-color>
`;

exports[`gux-form-field-color #render label-position should render component as expected (2) 1`] = `
<gux-form-field-color label-position="above">
  <mock:shadow-root>
    <div class="gux-above gux-form-field-container">
      <div class="gux-above gux-form-field-label">
        <slot name="label"></slot>
        <gux-form-field-label-indicator variant="required"></gux-form-field-label-indicator>
        <slot name="label-info"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-input-container">
            <slot name="input"></slot>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <input id="gux-form-field-input-i" slot="input" type="color" value="#ff0000">
  <label for="gux-form-field-input-i" slot="label">
    Label
  </label>
</gux-form-field-color>
`;

exports[`gux-form-field-color #render label-position should render component as expected (3) 1`] = `
<gux-form-field-color label-position="beside">
  <mock:shadow-root>
    <div class="gux-beside gux-form-field-container">
      <div class="gux-beside gux-form-field-label">
        <slot name="label"></slot>
        <gux-form-field-label-indicator variant="required"></gux-form-field-label-indicator>
        <slot name="label-info"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-input-container">
            <slot name="input"></slot>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <input id="gux-form-field-input-i" slot="input" type="color" value="#ff0000">
  <label for="gux-form-field-input-i" slot="label">
    Label
  </label>
</gux-form-field-color>
`;

exports[`gux-form-field-color #render label-position should render component as expected (4) 1`] = `
<gux-form-field-color label-position="screenreader">
  <mock:shadow-root>
    <div class="gux-form-field-container gux-screenreader">
      <div class="gux-form-field-label gux-screenreader">
        <slot name="label"></slot>
        <gux-form-field-label-indicator variant="required"></gux-form-field-label-indicator>
        <slot name="label-info"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-input-container">
            <slot name="input"></slot>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <input id="gux-form-field-input-i" slot="input" type="color" value="#ff0000">
  <label for="gux-form-field-input-i" slot="label">
    Label
  </label>
</gux-form-field-color>
`;

exports[`gux-form-field-color #render should render the label info when provided 1`] = `
<gux-form-field-color>
  <mock:shadow-root>
    <div class="gux-above gux-form-field-container">
      <div class="gux-above gux-form-field-label">
        <slot name="label"></slot>
        <gux-form-field-label-indicator variant="required"></gux-form-field-label-indicator>
        <slot name="label-info"></slot>
      </div>
      <div class="gux-input-and-error-container">
        <div class="gux-input">
          <div class="gux-input-container">
            <slot name="input"></slot>
          </div>
        </div>
        <div class="gux-form-field-error">
          <gux-icon decorative="" icon-name="fa/hexagon-exclamation-solid"></gux-icon>
          <div class="gux-message">
            <slot name="error"></slot>
          </div>
        </div>
        <div class="gux-form-field-help">
          <div class="gux-message">
            <slot name="help"></slot>
          </div>
        </div>
      </div>
    </div>
  </mock:shadow-root>
  <input aria-describedby="gux-label-info-beta-i" id="gux-form-field-input-i" name="e-1" slot="input" type="color" value="#cc3ebe">
  <label for="gux-form-field-input-i" slot="label">
    Default
  </label>
  <gux-label-info-beta id="gux-label-info-beta-i" slot="label-info">
    <span slot="content">
      This is some tooltip text
    </span>
  </gux-label-info-beta>
</gux-form-field-color>
`;
